app/tests/integration/nimbus/test_mobile_integration.py

```diff
@@ -0,0 +1,128 @@
+            android_sdk_version=base_app_context.android_sdk_version,
+            debug_tag=base_app_context.debug_tag,
+            installation_date=base_app_context.installation_date,
+            home_directory=base_app_context.home_directory,
+            custom_targeting_attributes=base_app_context.custom_targeting_attributes,
+        )
+
+    return _load_app_context_helper
+
+
+@pytest.fixture(name="ar_units")
+def fixture_ar_units():
+    return nimbus.AvailableRandomizationUnits(None, 0)
+
+
+@pytest.fixture(name="sdk_client")
+def fixture_sdk_client(ar_units):
+    def _client_helper(app_context):
+        return nimbus.NimbusClient(app_context, os.getcwd(), None, ar_units)
+
+    return _client_helper
+
+
+@pytest.fixture(name="jexl_evaluator")
+def fixture_jexl_evaluator():
+    def _eval_jexl(sdk_client, expression):
+        targetting_helper = sdk_client.create_targeting_helper()
+        try:
+            value = targetting_helper.eval_jexl(expression)
+        except nimbus.NimbusError.EvaluationError:
+            return None
+        else:
+            return value
+
+    return _eval_jexl
+
+
+@pytest.mark.parametrize("context", client_info_list())
+@pytest.mark.run_targeting
+def test_check_mobile_targeting(
+    jexl_evaluator, sdk_client, load_app_context, context, slugify, experiment_name
+):
+    context = json.loads(context["app_context"])
+    context["locale"] = context["locale"][:2]  # strip region
+    helpers.create_mobile_experiment(
+        slugify(experiment_name),
+        context["app_name"],
+        locale_number_loader([context["locale"]]),
+    )
+    data = helpers.load_experiment_data(slugify(experiment_name))
+    expression = data["data"]["experimentBySlug"]["jexlTargetingExpression"]
+    assert jexl_evaluator(sdk_client(load_app_context(context)), expression)
+
+
+@pytest.mark.parametrize("context", client_info_list())
+@pytest.mark.run_targeting
+def test_check_mobile_targeting_falsey(
+    jexl_evaluator, sdk_client, load_app_context, context, slugify, experiment_name
+):
+    context = json.loads(context["app_context"])
+    helpers.create_mobile_experiment(
+        slugify(experiment_name),
+        context["app_name"],
+        locale_number_loader([context["locale"][:2]]),
+    )
+    expression = "'zz' in locale"  # represents a malformed jexl expression
+    assert jexl_evaluator(sdk_client(load_app_context(context)), expression) is False
+
+
+@pytest.mark.run_targeting
+def test_check_mobile_targeting_undefined(
+    jexl_evaluator, sdk_client, load_app_context, slugify, experiment_name
+):
+    client = json.loads(client_info_list()[0]["app_context"])
+    helpers.create_mobile_experiment(
+        slugify(experiment_name),
+        client["app_name"],
+        locale_number_loader([client["locale"]]),
+    )
+    expression = ""  # represents a malformed jexl expression
+    assert jexl_evaluator(sdk_client(load_app_context(client)), expression) is None

```

Comment:
We don't need to test these cases here since that's covered by the SDK unit tests, the only thing we need to worry about here is the JEXL defined in the advanced targeting for mobile.

Reply:
Kinda what I figured but the acceptance criteria on the JIRA ticket mentioned these. Should I remove them?

Reply:
Malforming those will result in a `None` not a `False`

Reply:
Perfect then we can check for that similar to how we do for desktop 👍